src/cljs/nnts2/directory/components.cljs

```diff
@@ -0,0 +1,53 @@
+(ns nnts2.directory.components
+  (:require [reagent.core :as r]
+            [re-frame.core :as re-frame]
+            [nnts2.directory.subs :as subs]))
+
+
+(defn add-child-directory [dir]
+  (let [dir-details (r/atom {:name ""
+                             :org-id (:org-id dir)
+                             :parent-id (:id dir)})]
+    [:div {:class "row"}
+     [:div
+      {:class "column column-80"}
+      [:input {:type "text"
+               :style {:color "#FFFFFF"
+                       :margin-bottom 0}
+               :on-change (fn [e] (swap! dir-details assoc :name (-> e .-target .-value)))}]]
+     [:div
+      {:class "column column-10"
+       :style {:padding 0}}
+      [:button
+       {:style {:padding-left "5px"
+                :padding-right "5px"
+                :margin-bottom 0}
+        :on-click #(re-frame/dispatch [:nnts2.directory.events/create-directory-submit @dir-details])}
+       "→"]]
+     ])
+  )
+
+(defn directory [dir]
+  (let [selected (re-frame/subscribe [::subs/selected-directory (:id dir)])
+        add-new (r/atom false)]
+    (fn []
+      [:text
+       {:id (:id dir)
+        :onDoubleClick (fn [e] (swap! add-new not))
+        :on-click #(re-frame/dispatch [:nnts2.directory.events/directory-selected (:id dir)])
+        :style (if @selected {:color "orange"} {})}
+       (:name dir)
+       [:dl  [:dt (if (not@add-new) {:style  {:display "none"}}) [add-child-directory dir]]]])))
```

Comment:
What is `not@add-new`? How does that even work? Is it missing a space?

Reply:
its missing a space. will find out how it even compiled.